onnx_tf/handlers/backend/pool_mixin.py

```diff
@@ -115,6 +115,64 @@ def pool(cls, node, input_dict, pool_func, pooling_type, strict=True):
 
     return [pooled]
 
+  @classmethod
+  def pool_v10(cls, node, input_dict, pool_func, pooling_type, strict=True):
+    x = input_dict[node.inputs[0]]
+
+    kernel_shape = node.attrs["kernel_shape"]
+
+    spatial_size = len(kernel_shape)
+    x_rank = spatial_size + 2
+
+    kernel_shape = node.attrs["kernel_shape"]
+    strides = node.attrs.get("strides", [1] * spatial_size)
+    dilations = node.attrs.get("dilations", [1] * spatial_size)
+    ceil_mode = bool(node.attrs.get("ceil_mode", 0))
+    pads = node.attrs.get("auto_pad", "NOTSET")
+    if pads == "NOTSET":
+        pads = node.attrs.get("pads", [0] * spatial_size * 2)
+
+    if spatial_size > 3:
+      exception.OP_UNSUPPORTED_EXCEPT(
+          "MaxPool with {}D input".format(x_rank), "Tensorflow")
+    if pooling_type == "MAX_WITH_ARGMAX" and x_rank != 4:
+      exception.OP_UNSUPPORTED_EXCEPT(
+          "MaxPool with {}D input".format(x_rank), "Tensorflow")
+    if node.attrs.get("storage_order", 0) != 0:
+      exception.OP_UNSUPPORTED_EXCEPT("MaxPool with column major",
+                                      "Tensorflow")
+
+    storage_format, _ = get_data_format(x_rank)
+
+    need_trans = storage_format.startswith("NC")
+    if need_trans:
+      compute_format = "N" + storage_format[2:] + "C"
+      x = tf.transpose(x, perm=get_perm_from_formats(storage_format,
+                                                     compute_format))
+
+    dp = DilatedPooling(input=x, kernel_shape=kernel_shape, strides=strides,
+                        dilations=dilations, padding=pads, ceil_mode=ceil_mode)
+
+    result = []
+    if pooling_type == "MAX":
```

Comment:
I'm curious, any chance we can simplify this cross product here? pooling type x need_trans?
Something like this:
preprocess_op = tf.transpose... if need_trans else tf.identity...
pool_op = dp. dilated_maxpool if pooling_type == "MAX" else dilated_maxpool_with_argmax

unpacking results may be tricky, please let me know if you think its improvable.

Reply:
good idea. had to use lambda functions.